mapper/orm/src/main/java/org/hibernate/search/mapper/orm/session/impl/HibernateOrmSearchSessionHolder.java

```diff
@@ -1,111 +0,0 @@
-/*
- * SPDX-License-Identifier: Apache-2.0
- * Copyright Red Hat Inc. and Hibernate Authors
- */
-package org.hibernate.search.mapper.orm.session.impl;
-
-import java.io.Serializable;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
-
-import jakarta.transaction.Synchronization;
-
-import org.hibernate.Transaction;
-import org.hibernate.engine.spi.SessionImplementor;
-import org.hibernate.search.mapper.pojo.work.spi.PojoIndexingPlan;
-
-public class HibernateOrmSearchSessionHolder implements Serializable {
-
-	private static final String SESSION_PROPERTY_KEY = "hibernate.search.session";
-
-	private static final Map<Transaction, HibernateOrmSearchSessionHolder> holderPerClosedSessionTransaction =
-			new ConcurrentHashMap<>();
-
-	// Public for tests only
-	public static int staticMapSize() {
-		return holderPerClosedSessionTransaction.size();
-	}
-
-	public static HibernateOrmSearchSessionHolder get(SessionImplementor session, boolean createIfMissing) {
-		HibernateOrmSearchSessionHolder holder =
-				(HibernateOrmSearchSessionHolder) session.getProperties().get( SESSION_PROPERTY_KEY );
-		if ( holder != null ) {
-			return holder;
-		}
-		boolean closedSessionAndTransaction = session.isClosed() && session.isTransactionInProgress();
-		if ( closedSessionAndTransaction ) {
-			// This can happen when using JTA.
-			// Only do this as a fallback: if somehow the session holder was added to the session properties
-			// before the session was closed, we definitely want to use it and avoid the static map.
-			holder = holderPerClosedSessionTransaction.get( session.accessTransaction() );
-		}
```

Comment:
Hi @yrodiere 👋🏻 🙂 

I've tried to integrate the ORM extension, and it made me wonder if this ^ could even happen with holder being attached to a session as soon as it (session) is created.
The tests passed without this static map.. but ... I'm not sure whether we actualy have any tests to hit this scenario, or we do?

Reply:
Hey. It's been a long time... but I don't see a reason this would still be needed if the holder is added when creating the session.

Also, the PR that added this code seems to be specifically about closed sessions: https://github.com/hibernate/hibernate-search/pull/2450
So if the tests from that PR are still around, and still pass... +1 to remove all this code?